app/results.tsx

```diff
@@ -69,10 +81,29 @@ export function Results() {
     fetchAsync();
   }, []);
 
-  const filteredResults = results.filter((result) => {
-    if (!q) return true;
-    return result.name.includes(q) || result.id.includes(q);
-  });
+  useEffect(() => {
+    const sortMethodObj = sortMethod === 'alpha-asc' ? alphaAscendingSortMethod :
+      sortMethod === 'alpha-desc' ? alphaDescendingSortMethod :
+      sortMethod === 'date-new' ? createdAtNewestFirstSortMethod :
+      createdAtOldestFirstSortMethod;
+
+    const sortedResults = 
+      (sortMethodObj.attribute === 'created_at') ?
+        results.sort((a, b) =>
+          (a.created_at > b.created_at ? 1 : -1 ) * Math.sign(sortMethodObj.direction)
+        ) :
+        results.sort((a, b) =>
+          (a.id > b.id ? 1 : -1) * Math.sign(sortMethodObj.direction)
+        );
+
+    setFilteredResults(sortedResults.filter((result) => {
```

Comment:
I think it makes sense to filter and then sort, for performance

Reply:
Doing this via setstate and not just a var like previously is increasing input lag somehow

Reply:
I agree about the input lag, which appears to get worse with more rows. The commit below attempts to limit the number of rows rendered, by using @tanstack/react-table